server/src/server.rs

```diff
@@ -345,6 +347,78 @@ impl Server {
         Ok(())
     }
 
+    async fn reverse_request(
+        &self,
+        request_id: RequestId,
+        session_id: SessionId,
+        from: SocketAddr,
+        params: proto::request::ReverseConnection,
+    ) -> ServerResult<()> {
+        let source_node_info = match self.state.read().await.nodes.get_by_session(session_id) {
+            None => return Err(Unauthorized::SessionNotFound(session_id).into()),
+            Some(node_id) => node_id.info,
+        };
+        if source_node_info.endpoints.is_empty() {
+            return Err(BadRequest::NoPublicEndpoints.into());
+        }
+
+        let target_node_id = (&params.node_id)
+            .try_into()
+            .map_err(|_| BadRequest::InvalidNodeId)?;
+        let target_session = {
+            match self.state.read().await.nodes.get_by_node_id(target_node_id) {
+                None => {
+                    return Err(InternalError::Generic(format!(
+                        "There is no session with node_id {}",
+                        target_node_id
+                    ))
+                    .into())
+                }
+                Some(session) => session,
+            }
+        };
+
+        let message_to_send = proto::Packet::control(
+            target_session.session.to_vec(),
+            proto::control::ReverseConnection {
+                node_id: source_node_info.node_id.into_array().to_vec(),
+                endpoints: source_node_info
+                    .endpoints
+                    .into_iter()
+                    .map(proto::Endpoint::from)
+                    .collect(),
+            },
+        );
+        self.send_to(message_to_send, &target_session.address)
+            .await
+            .map_err(|_| InternalError::Send)?;
+
+        let reponse = proto::Packet::response(
+            request_id,
+            session_id.to_vec(),
+            StatusCode::Ok,
+            proto::response::ReverseConnection {},
+        );
+
+        self.send_to(reponse, &from)
+            .await
+            .map_err(|_| InternalError::Send)?;
+
+        log::info!(
+            "ReverseConnection sent. source: {}, target: {}, request: {}",
+            &from,
+            &target_session.address,
+            &request_id,
+        );
+        log::debug!(
+            "source_session: {}, target_session: {}",
```

Comment:
```suggestion
            "ReverseConnection - source_session: {}, target_session: {}",
```

Reply:
this line is extra info for the previous log line and should always show up together